protocols/dcutr/examples/client.rs

```diff
@@ -197,25 +197,41 @@ fn main() -> Result<(), Box<dyn Error>> {
         }
     }
 
-    // Wait till connected to relay to learn external address.
+    // Wait till connected to relay to learn external address. In case we are in listening mode,
+    // wait for the relay to accept our reservation request.
     block_on(async {
+        let mut learned_observed_addr = false;
+        let mut relay_accepted_reservation = false;
+
         loop {
             match swarm.next().await.unwrap() {
                 SwarmEvent::NewListenAddr { .. } => {}
                 SwarmEvent::Dialing { .. } => {}
                 SwarmEvent::ConnectionEstablished { .. } => {}
                 SwarmEvent::Behaviour(Event::Ping(_)) => {}
+                SwarmEvent::Behaviour(Event::Relay(client::Event::ReservationReqAccepted {
+                    ..
+                })) => {
+                    info!("Relay accepted our reservation request.");
+                    relay_accepted_reservation = true
+                }
                 SwarmEvent::Behaviour(Event::Relay(_)) => {}
```

Comment:
Maybe also catch the case of `client::Event::ReservationReqFailed` and return an error in that case. 
Shouldn't happen in practice but I'd rather have an explicit error than a non-terminating program. 
Alternatively, I think there is no valid case in which any other relay event should occur anyway/ So we could remove the match block here and let it fall to the blank panic below.